src/Psalm/Config.php

```diff
@@ -1734,6 +1734,16 @@ public function visitStubFiles(Codebase $codebase, ?Progress $progress = null):
             $core_generic_files[] = $stringable_path;
         }
 
+        if (!\extension_loaded('ibm_db2')) {
```

Comment:
I don't really understand why the rest of the conditions look like `if (extension_loaded(/* ... */))`. As I understand, one of the purposes of the stubs is to enable the static analysis of the code that depends on an extension that is unavailable at runtime. Loading the stub only when the extension is loaded doesn't help to solve the issue.

Reply:
> one of the purposes of the stubs is to enable the static analysis of the code that depends on an extension that is unavailable at runtime.

Oh, in that case you should instead provide your own stubs. If Psalm were to stub all classes it might miss situations where you're missing an extension.

Reply:
I think I don't understand the architecture of stubs in Psalm. There are more or less complete stubs from JetBrains but they are released quite rarely and I want to try to get rid of them given that Psalm and PHPStan provide their own. Having to have 3 sources of stubs, each of which may have its issues is quite a burden.

Reply:
Psalm needs them to provide more detailed types that are unavailable through reflection. I think you can still force-load them in your `psalm.xml`, e.g.:
```
<stubs>
   <file name="vendor/vimeo/psalm/stubs/ext-ibm-db2.phpstub"/>
</stubs>
```

Reply:
> Psalm needs them to provide more detailed types that are unavailable through reflection.

Right, but Psalm also uses the call maps that provide some basic information on the extension API. As a result, it can analyze most of the code that uses the extension (e.g. calls the functions) but not all of it (the usage of constants). From the end-user standpoint, this doesn't look right.